public/client.js

```diff
@@ -0,0 +1,262 @@
+
+var log = console.log.bind(console), keyData = document.getElementById('key_data'),
```

Comment:
I've read that this one line use of `var` is a matter of personal preference, but I wonder if it would be a better convention to simply make each `var` in its own line. Not a big deal, but it seems like it might make the code easier to read. Something to consider!